### ISSUE TITLE
Rule CV12 fix deletes WHERE clause when joined table uses Jinja templating

### ISSUE THREAD

**Issue by ronkot** [sqlfluff/sqlfluff#8230]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When the joined table reference uses a Jinja template expression (e.g. `{{ ref('table') }}`), `sqlfluff fix` with rule CV12 **deletes the entire WHERE clause** instead of moving its conditions to an ON clause. This results in silently corrupted SQL that produces wrong results.

Without Jinja templating on the joined table, the fix works correctly (moves WHERE conditions to ON).

### Expected Behaviour

```sql
-- input.sql
select a.id, b.id as b_id
from table_a as a
left join {{ ref('table_b') }} as b
where a.region = b.region
```

```bash
sqlfluff fix --dialect snowflake --rules CV12 input.sql
```

### Expected result:
```sql
select a.id, b.id as b_id
from table_a as a
left join {{ ref('table_b') }} as b
    on a.region = b.region
```

### Observed Behaviour

```sql
-- input.sql
select a.id, b.id as b_id
from table_a as a
left join {{ ref('table_b') }} as b
```

### How to reproduce

```sql
-- input.sql
select a.id, b.id as b_id
from table_a as a
left join {{ ref('table_b') }} as b
where a.region = b.region
```

```bash
sqlfluff fix --dialect snowflake --rules CV12 input.sql
```
--> see how the `where` is dropped and no `on` is added

### Dialect

snowflake

### Version

4.2.2

### Configuration

[sqlfluff]
dialect = snowflake
templater = jinja

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
